dbt/adapters/athena/impl.py

```diff
@@ -786,3 +787,21 @@ def _generate_snapshot_migration_sql(self, relation: AthenaRelation, table_colum
                 drop_staging_sql.strip(),
             ]
         )
+
+    @available
+    def get_table_type(self, db_name, table_name) -> TableType:
```

Comment:
we already have this method in `relation` module. Seems that we don't need such code duplication. Does it make sense to move this method to `AthenaRelation` class if you need it to be used inside macro ? @Jrmyy WDYT ?

Reply:
Hey,
Multiple things here:
1. We have a standard now that required, whenever possible, to pass the relation object to the function instead of db_name and table_name
2. We already do something similar inside `get_glue_table_location`. I would suggest to find a way to reuse part of this code, or at least to prevent from duplication

Reply:
Refactoring completed, see latest. Unit/func tests all passing for me